src/Microsoft.AspNet.HttpOverrides/OverrideHeaderMiddleware.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.HttpOverrides
+{
+    public class OverrideHeaderMiddleware
+    {
+        private const string XForwardedFor = "X-Forwarded-For";
+        private const string XForwardedHost = "X-Forwarded-Host";
+        private const string XForwardedProto = "X-Forwarded-Proto";
+        private readonly OverrideHeaderMiddlewareOptions _options;
+        private readonly RequestDelegate _next;
+
+        public OverrideHeaderMiddleware(RequestDelegate next, OverrideHeaderMiddlewareOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(_next));
+            }
+            _options = options;
+            _next = next;
+        }
+
+        public Task Invoke(HttpContext context)
+        {
+            if ((_options.ForwardedOptions & ForwardedHeaders.XForwardedFor) != 0)
+            {
+                var xForward = context.Request.Headers.GetCommaSeparatedValues(XForwardedFor);
+                if (xForward != null && xForward.Count > 0)
+                {
+                    IPAddress originalIPAddress;
+                    if (IPAddress.TryParse(xForward[0], out originalIPAddress))
+                    {
+                        context.Connection.RemoteIpAddress = originalIPAddress;
+                    }
+                }
+            }
+
+            if ((_options.ForwardedOptions & ForwardedHeaders.XForwardedHost) != 0)
+            {
+                var xForwardHost = context.Request.Headers.Get(XForwardedHost);
+                if (xForwardHost != null)
+                {
+                    context.Request.Host = new HostString(xForwardHost);
+                }
+            }
+
+            if ((_options.ForwardedOptions & ForwardedHeaders.XForwardedProto) != 0)
+            {
+                var xForwardProto = context.Request.Headers.Get(XForwardedProto);
+                if (XForwardedProto != null)
```

Comment:
You mixed up the constant name and the variable name. Use `xForwardProto` here or you'll get a nasty surprise :smile: